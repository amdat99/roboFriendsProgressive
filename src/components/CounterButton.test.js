import { shallow } from 'enzyme';
import React from 'react';
import CounterButton from './CounterButton'

import {createSerializer} from 'enzyme-to-json';
import toJson from 'enzyme-to-json';

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

	it('expect to render CounterButton component', () => {
		const mockColor = 'red'
	// Method1
	expect(toJson(shallow(<CounterButton color={mockColor} />))).toMatchSnapshot();
	})

it('correnctly increments the counter', () => {
  const wrapper = shallow(<CounterButton/>)

  expect(wrapper).toMatchSnapshot();
      expect((wrapper.state())).toEqual({count: 0})
  	wrapper.find('[id="counter"]').simulate('click');
    wrapper.find('[id="counter"]').simulate('click');
    expect((wrapper.state())).toEqual({count: 2})
    wrapper.find('[id="counter"]').simulate('keypress');
    expect((wrapper.state())).toEqual({count: 2})
    
 
    });

  it('shouldComponentUpdate returns false when values match', () => {
  const wrapper = shallow(<CounterButton/>)
  wrapper.find('[id="counter"]').simulate('click');
   expect(wrapper.instance().shouldComponentUpdate({count: 1},{count: 1})).toBe(false)
 })

