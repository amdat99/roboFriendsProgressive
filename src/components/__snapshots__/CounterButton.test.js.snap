// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correnctly increments the counter 1`] = `
<div>
  <button
    id="counter"
    onClick={[Function]}
  >
    count: 
    0
  </button>
</div>
`;

exports[`expect to render CounterButton component 1`] = `
<div>
  <button
    color="red"
    id="counter"
    onClick={[Function]}
  >
    count: 
    0
  </button>
</div>
`;
